azure-pipelines.yml

```diff
@@ -49,7 +49,8 @@ jobs:
                  eq(dependencies.check_for_release.outputs['out.is_release'], 'true'),
                  eq(variables['Build.SourceBranchName'], 'main'))
   pool:
-    vmImage: "Ubuntu-16.04"
+    name: ubuntu_20_04
+    demands: assignment -equals default
```

Comment:
We don't set assignments on Azure-hosted VMs, so it may fail to find any machine to run this.

Reply:
🤦 you're giving up on Azure nodes.

Reply:
this is not an azure hosted vm. I don’t really see a reason why this should run on one so I tried to switch to our own vms at the same time.

Reply:
I reverted this part for now. imho we should try to switch those as well but I’d like to spend at least some time going over what we do here exactly and convincing myself that this is not going to fail on the next release and the ghc-lib stuff has turned out to be more complex than expected so doing it separately seems sensible.